src/ICS26Router.sol

```diff
@@ -147,6 +148,22 @@ contract ICS26Router is IICS26Router, IICS26RouterErrors, Ownable, ReentrancyGua
 
         ICS02_CLIENT.getClient(msg_.packet.destChannel).membership(membershipMsg);
 
+        // recvPacket will no-op if the packet receipt already exists
+        // solhint-disable-next-line no-empty-blocks
+        try IBC_STORE.setPacketReceipt(msg_.packet) { }
+        catch (bytes memory reason) {
+            if (bytes4(reason) == IICS24HostErrors.IBCPacketReceiptAlreadyExists.selector) {
```

Comment:
I kind of dislike these long pieces of code in the middle of the core ibc handling. I think it could be worth it to bring the catch part down to a single function call, something like:
```solidity
catch (bytes memory reason) {
  return NoopOn(reason, IICS24HostErrors.IBCPacketReceiptAlreadyExists.selector);
}
```
This could be reused with the other ones as well, obviously.

Reply:
I tried this and the tests fail. Probably due to assembly code revert shenanigans. I'll not do this at the moment. We can revisit this if we find a way?

Reply:
Ok, fair enough :)

Reply:
Nevermind, I got it to work I think. I'll push this